gdrive/idva/pivot_director.py

```diff
@@ -0,0 +1,274 @@
+from gdrive.sheets.builders import FormulaBuilder, PivotTableBuilder
+from gdrive.sheets.types import (
+    SortOrderEnum,
+    FilterTypeEnum,
+    FormulaEnum,
+    UserEnteredValue,
+    SummarizeFunctionEnum,
+    StringLiteral,
+)
+
+
+class IDVAPivotDirector:
+    def clicks(self, col_dict: dict) -> dict:
+        builder = PivotTableBuilder(0, col_dict)
+        builder.add_row("eventName", sortOrder=SortOrderEnum.ASCENDING)
+        builder.add_row("eventCount", sortOrder=SortOrderEnum.ASCENDING)
+        builder.add_value("eventCount", SummarizeFunctionEnum.SUM)
+        return builder.render()
+
+    def facebook(self, col_dict: dict) -> dict:
+        builder = PivotTableBuilder(0, col_dict)
+        builder.add_row("eventName", SortOrderEnum.ASCENDING, show_totals=False)
+        builder.add_row("firstUserMedium", SortOrderEnum.ASCENDING)
+        builder.add_row("firstUserSource", SortOrderEnum.ASCENDING)
+        builder.add_row(
+            "firstUserCampaignName", SortOrderEnum.ASCENDING, show_totals=False
+        )
+
+        builder.add_value("eventCount", SummarizeFunctionEnum.SUM)
+
+        # =OR(regexmatch(firstUserSource,"facebook"),regexmatch(firstUserSource,"fb.com"), regexmatch(firstUserMedium,"fb"))
+        facebook = FormulaBuilder(
```

Comment:
I am not sure I understand the purpose of the formula builder. The formula string seems like a better representation than the code.

Reply:
So writing a Formula this way gives a few advantages. 

1. Speeds up writing the formula with intellisense
2. Allows for the linter to detect errors and avoid needing to debug syntax errors by hand on the finished sheet
3. Simplifies the development and maintenance of nested functions in particular
4. This is a fringe benefit and is not being used here, but it would allow for flexibility at runtime to move components in and out to avoid having to recreate the instance every time

In this use case, the formula isnt terribly complicated but if there were another layer of nesting maintaining this formula would become a pain really quick.